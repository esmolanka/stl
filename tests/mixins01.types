
type Foo = mixin
  { foo : Int
  }

type Bar = mixin
  { bar : Int
  }

type Baz = mixin
  { baz : Float
  , mix Foo
  , mix Bar
  }

type Rec1 =
  { mix Foo
  , mix Bar
  }

type Rec2 =
  { foo : Int
  , bar : Int
  }

#check Rec1 <: Rec2
#check {mix Baz} <: Rec1
