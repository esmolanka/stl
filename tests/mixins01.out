tests/mixins01.types:26:1-26:20:
  Subtype:
    ∃ (ρ : Row). {foo : Int, bar : Int | ρ}
  Supertype:
    ∃ (ρ : Row). {foo : Int, bar : Int | ρ}
  Result:
    OK
tests/mixins01.types:27:1-27:25:
  Subtype:
    ∃ (ρ : Row). {baz : Float, foo : Int, bar : Int | ρ}
  Supertype:
    ∃ (ρ : Row). {foo : Int, bar : Int | ρ}
  Result:
    OK